assets/js/index.js

```diff
@@ -0,0 +1,146 @@
+  {
+    timeLabel: "09:00AM",
+    key: 9,
+  },
+  {
+    timeLabel: "10:00AM",
+    key: 10,
+  },
+  {
+    timeLabel: "11:00AM",
+    key: 11,
+  },
+  {
+    timeLabel: "12:00AM",
+    key: 12,
+  },
+  {
+    timeLabel: "13:00PM",
+    key: 13,
+  },
+  {
+    timeLabel: "14:00PM",
+    key: 14,
+  },
+  {
+    timeLabel: "15:00PM",
+    key: 15,
+  },
+  {
+    timeLabel: "16:00PM",
+    key: 16,
+  },
+  {
+    timeLabel: "17:00PM",
+    key: 17,
+  },
+];
+
+//~ LOCAL STORAGE
+
+const readFromLocalStorage = (key) => {
+  // get from LS using key name
+  const dataFromLS = localStorage.getItem(key);
+
+  // parse data from LS
+  const parsedData = JSON.parse(dataFromLS);
+  if (parsedData) {
+    return parsedData;
+  } else {
+    return "";
+  }
+};
+
+const writeToLocalStorage = (key, value) => {
+  // convert value to string
+  const stringifiedValue = JSON.stringify(value);
+
+  // set stringified value to LS for key name
+  localStorage.setItem(key, stringifiedValue);
+};
+
+//~ FUNCTIONS SECTION
+
+// function to create HTML for each time block, append to page
+const renderTimeBlocks = () => {
+  // target the time block container and create time blocks for each working hour in workingHours array
+  $.each(workingHours, function (index, workingHour) {
+    // get current hour using moment js
+    const currentHour = moment().hour();
+    // function to render the textarea color depending on the conditional statements
+    const renderTextareaColor = () => {
+      // if current hour is equal to working hour
+      if (currentHour === workingHour.key) {
+        return "present";
+      }
+      //if current hour is less than working Hour
+      else if (currentHour < workingHour.key) {
+        return "future";
+        //if current hour greater than  working Hour
+      } else if (currentHour > workingHour.key) {
+        return "past";
+      }
+    };
+
+    // step 1: looping through all the working hours array and appending each working hour to time block container
+    // step 2: calling fn render text area color inside <textarea> html element to run fn for each loop and return text color based on condition
+    $(
+      "#time-block-container"
+    ).append(`<div class="time-blocks d-flex flex-row align-items-center">
+        <div id=timeLabel">${workingHour.timeLabel}</div>
+        <textarea
+          class="text-area form-control ${renderTextareaColor()}"
+          id="floatingTextarea2"
+          style="height: 100px"
+        >${readFromLocalStorage(workingHour.key)}</textarea>
+        <div class="button-container">
+          <button class="save-button">save</button>
+        </div>
+        </div>`);
+  });
+  // targeting all buttons
+  const buttons = document.querySelectorAll(".button-container");
+  // targeting text area elements
+  const textAreas = document.querySelectorAll(".text-area");
+  // looping trough each button and attaching an event listener
+  buttons.forEach((btn, index) => {
```

Comment:
You could have used event delegation here to add a click event listener on the element that contains all time blocks and then checked to see if a click originated from a button and then get the `id` of the button that was clicked to determine which time block to save the value in. This is not performant as the document has too many event listeners registered on page load. This is also generally bad practice since if you have more buttons then there are far more event listeners on the page upon load.